### PR TITLE
Fix PGP command line user interface

### PR DESCRIPTION
**Pull request by patrickxb**
Previously, Identify/Track was handled by PGPKeyfinder.  This led to some issues, especially on decrypt/verify of self-signed message.

I moved it out and encrypt/decrypt handle it.  Also cleaned up command line flags.  Some were not used, some were just wrong.

This behavior might change as the Identify system changes, but it should be more flexible in this state.

Also, created a PgpUI to output the signature success message instead of using LogUI.

There's a big test of the UI in engine/pgp_test.go, TestPGPUserInterface.

### CODE REVIEW

**Comment by maxtaco** [comment]
@patrickxb see C/I failures above

File: go/engine/pgp_common.go [hunk 1]
```diff
@@ -6,16 +6,28 @@ package engine
...
-		ctx.LogUI.Notice("Signature verified. Signed by %s %s (%s).", owner.GetName(), humanize.Time(signatureTime), signatureTime)
+	arg := keybase1.OutputSignatureSuccessArg{
+		Fingerprint: fingerprint.String(),
+		Username:    owner.GetName(),
+		SignedAt:    keybase1.TimeFromSeconds(signatureTime.Unix()),
 	}
-	ctx.LogUI.Notice("PGP Fingerprint: %s.", fingerprint)
+	ctx.PgpUI.OutputSignatureSuccess(context.TODO(), arg)
+
+	/*
```
Comment (maxtaco) [inline go/engine/pgp_common.go]: Fine to nuke this code.

Comment (patrickxb) [inline go/engine/pgp_common.go]: done

**Comment by patrickxb** [comment]
Fixed systests.  CI should be fine now.

File: go/client/pgp_ui.go [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+func NewPgpUIProtocol(g *libkb.GlobalContext) rpc.Protocol {
+	return keybase1.PGPUiProtocol(g.UI.GetPgpUI())
+}
+
+func (p PgpUI) OutputSignatureSuccess(_ context.Context, arg keybase1.OutputSignatureSuccessArg) error {
+	signedAt := keybase1.FromTime(arg.SignedAt)
+	if signedAt.IsZero() {
+		p.parent.Printf("Signature verified. Signed by %s.\n", arg.Username)
+	} else {
+		p.parent.Printf("Signature verified. Signed by %s %s (%s).\n", arg.Username, humanize.Time(signedAt), signedAt)
```
Comment (maxtaco) [inline go/client/pgp_ui.go]: what about that feature "signed by **you**"?

**Comment by patrickxb** [comment]
LGTM

**Comment by maxtaco** [comment]
hmm we have conflicts now

**Pull request merged**